### PR TITLE
chore(uniffi): Update `uniffi` to another branch

### PR DESCRIPTION
**Pull request by Hywan**
This patch switches UniFFI from
https://github.com/mozilla/uniffi-rs/pull/1684 to
https://github.com/mozilla/uniffi-rs/pull/1697.

Tested by @ganfra, and it works.

### CODE REVIEW

**Review by poljar — APPROVED** [review]
We should wait for the upstream Uniffi PR to be merged before we merge this one.

**Comment by bnjbvr** [comment]
This broke generation of swift bindings (including in our CI task that failed above), so reverting in a new PR.

**Pull request merged**
